src/main/java/com/epam/esm/dao/service/impl/PersistenceServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.epam.esm.dao.service.impl;
+
+import com.epam.esm.dao.service.PersistenceService;
+import com.epam.esm.dao.sort.SortBy;
+import com.epam.esm.dao.sort.SortType;
+import com.epam.esm.model.BaseModel;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.Tag;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.Set;
+
+@Component
+@Scope("prototype")
```

Comment:
It's not a service (now renamed). It is a manager to work with DB, and this class work with generic entity. So, if we create this class only one time, it will work only with 1 type of entity (for example, only for user). That's why prototype needed here